src/App.tsx

```diff
@@ -15,48 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum Sort {
+  Default = '',
+  Length = 'length',
+  Alphabet = 'name',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case Sort.Alphabet:
+          return a.localeCompare(b);
+        case Sort.Length:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? preparedGoods.reverse() : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
```

Comment:
In your enum 'Sort' you should have default type of useState initial value 
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/74732843/baa2d7f8-7e4f-460d-8585-f19540b0a770)
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/74732843/4e7766a0-d202-445a-9c00-62b45f44bd84)

Reply:
Good point, fix it